torch/distributed/fsdp/api.py

```diff
@@ -381,7 +387,21 @@ class LocalOptimStateDictConfig(OptimStateDictConfig):
 
 @dataclass
 class ShardedOptimStateDictConfig(OptimStateDictConfig):
-    pass
+    """
+    ``ShardedOptimStateDictConfig`` is a config class meant to be used with
+    ``StateDictType.SHARDED_STATE_DICT``.
+
+    Attributes:
+        _use_dtensor (bool): If ``True``, then FSDP saves the state dict values
+            as ``DTensor``, and if ``False``, then FSDP saves them as
+            ``ShardedTensor``. (Default: ``False``)
+
+    .. warning:: `_use_dtensor` is a private field of ShardedOptimStateDictConfig and
+      it is used by FSDP to determine the type of state dict values. Users should not
+      manually modify `_use_dtensor`.
```

Comment:
<img width="808" alt="Screenshot 2023-08-25 at 5 39 48 PM" src="https://github.com/pytorch/pytorch/assets/31054793/3d81085e-6c06-4e4a-bd24-b3854d906d15">

nit: I think?
```suggestion
    .. warning:: ``_use_dtensor`` is a private field of :class:`ShardedOptimStateDictConfig` and
      it is used by FSDP to determine the type of state dict values. Users should not
      manually modify ``_use_dtensor``.
```